frontend/src/modules/Vfp/utils/VfpDataAccessor.ts

```diff
@@ -109,60 +118,64 @@ export class VfpDataAccessor {
-            const startIndex = nbFlowRates*(nbAlqValues*(nbGfrValues*(nbWfrValues*thpIndex+wfrIndex)+gfrIndex)+alqIndex)
-            return this._vfpTable.bhp_values.slice(startIndex, startIndex+nbFlowRates)
+        if (!("isProdTable" in this._vfpTable)) { 
+            throw Error("The getVfpProdBhpValues function is only valid for Production VFP tables.")            
         }
-        throw Error("The getVfpProdBhpValues function is not valid for VFPINJ tables.")
+        const nbWfrValues = this._vfpTable.wfrValues.length
+        const nbGfrValues = this._vfpTable.gfrValues.length
+        const nbAlqValues = this._vfpTable.alqValues.length
+        const nbFlowRates = this._vfpTable.flowRateValues.length
+        const startIndex = nbFlowRates*(nbAlqValues*(nbGfrValues*(nbWfrValues*thpIndex+wfrIndex)+gfrIndex)+alqIndex)
+        return this._vfpTable.bhpValues.slice(startIndex, startIndex+nbFlowRates)
     }
 
     getVfpInjBhpValues(thpIndex: number) : number[] {
-        if (this._vfpTable.vfp_type === VfpType_api.VFPINJ) {
-            const nbFlowRates = this._vfpTable.flow_rate_values.length
-            const startIndex = nbFlowRates*thpIndex
-            return this._vfpTable.bhp_values.slice(startIndex, startIndex+nbFlowRates)
+        if(!("isInjTable" in this._vfpTable)) {
+            throw Error("The getVfpInjBhpValues function is only valid for Injection VFP tables.")    
         }
-        throw Error("The getVfpInjBhpValues function is not valid for VFPPROD tables.")    
+        const nbFlowRates = this._vfpTable.flowRateValues.length
+        const startIndex = nbFlowRates*thpIndex
+        return this._vfpTable.bhpValues.slice(startIndex, startIndex+nbFlowRates)
     }
 
     getNumberOfValues(vfpParam: VfpParam): number {
         if (vfpParam == VfpParam.THP) {
-            return this._vfpTable.thp_values.length
+            return this._vfpTable.thpValues.length
         } 
-        if ("wfr_values" in this._vfpTable) { // This means that it is a VFPPROD table
-            if (vfpParam == VfpParam.WFR) {
-                return this._vfpTable.wfr_values.length
-            } else if (vfpParam == VfpParam.GFR) {
-                return this._vfpTable.gfr_values.length
-            } else if (vfpParam == VfpParam.ALQ) {
-                return this._vfpTable.alq_values.length
+        if (isProdTable(this._vfpTable)) { // This means that it is a VFPPROD table
+            if (vfpParam === VfpParam.WFR) {
+                return this._vfpTable.wfrValues.length
+            } else if (vfpParam === VfpParam.GFR) {
+                return this._vfpTable.gfrValues.length
+            } else if (vfpParam === VfpParam.ALQ) {
+                return this._vfpTable.alqValues.length
             }            
         }
-        return NaN
+        throw Error("Unhandled vfp parameter ${vfpParam}")
     }
 }
+
+export function isProdTable(vfpTable: VfpProdTable_api | VfpInjTable_api): vfpTable is VfpProdTable_api {
```

Comment:
Consider placing the function in a separate file in the `utils`-folder. Name it e.g. `vfpTableClassifier.ts`